test/acceptance/replication/replica_replication/multitenancy_test.go

```diff
@@ -0,0 +1,482 @@
+			Enabled:              true,
+		}
+		paragraphClass.Vectorizer = "text2vec-contextionary"
+		helper.CreateClass(t, paragraphClass)
+	})
+
+	t.Run("create tenant", func(t *testing.T) {
+		tenants := []*models.Tenant{{Name: tenantName, ActivityStatus: "HOT"}}
+		helper.CreateTenants(t, paragraphClass.Class, tenants)
+	})
+
+	t.Run("insert paragraphs", func(t *testing.T) {
+		batch := make([]*models.Object, len(paragraphIDs))
+		for i, id := range paragraphIDs {
+			batch[i] = articles.NewParagraph().
+				WithID(id).
+				WithContents(fmt.Sprintf("paragraph#%d", i)).
+				WithTenant(tenantName).
+				Object()
+		}
+		common.CreateObjectsCL(t, compose.GetWeaviate().URI(), batch, types.ConsistencyLevelAll)
+	})
+
+	var sourceNode, targetNode string
+	var targetNodeID int
+
+	t.Run("find source and target nodes", func(t *testing.T) {
+		verbose := verbosity.OutputVerbose
+		params := nodes.NewNodesGetClassParams().WithOutput(&verbose).WithClassName(paragraphClass.Class)
+		body, clientErr := helper.Client(t).Nodes.NodesGetClass(params, nil)
+		require.NoError(t, clientErr)
+		require.NotNil(t, body.Payload)
+
+		for i, node := range body.Payload.Nodes {
+			if len(node.Shards) == 0 {
+				targetNode = node.Name
+				targetNodeID = i + 1
+			} else {
+				sourceNode = node.Name
+			}
+		}
+
+		require.NotEmpty(t, sourceNode)
+		require.NotEmpty(t, targetNode)
+	})
+
+	t.Run("deactivate tenant", func(t *testing.T) {
+		tenants := []*models.Tenant{{Name: tenantName, ActivityStatus: "COLD"}}
+		helper.UpdateTenants(t, paragraphClass.Class, tenants)
+	})
+
+	transferType := api.COPY.String()
+
+	var uuid strfmt.UUID
+
+	t.Run(fmt.Sprintf("start %q of %s from %s to %s for paragraph", transferType, tenantName, sourceNode, targetNode), func(t *testing.T) {
+		resp, err := helper.Client(t).Replication.Replicate(
+			replication.NewReplicateParams().WithBody(
+				&models.ReplicationReplicateReplicaRequest{
+					CollectionID:        &paragraphClass.Class,
+					SourceNodeName:      &sourceNode,
+					DestinationNodeName: &targetNode,
+					ShardID:             &tenantName,
+					TransferType:        &transferType,
+				},
+			),
+			nil,
+		)
+		require.NoError(t, err)
+		require.Equal(t, http.StatusOK, resp.Code(), "replication replicate operation didn't return 200 OK")
+		uuid = *resp.Payload.ID
+	})
+
+	t.Run("waiting for replication to finish", func(t *testing.T) {
+		require.EventuallyWithT(t, func(ct *assert.CollectT) {
+			details, err := helper.Client(t).Replication.ReplicationDetails(
+				replication.NewReplicationDetailsParams().WithID(uuid), nil,
+			)
+			require.Nil(ct, err, "failed to get replication details %s", err)
+
+			if !assert.Equal(ct, "READY", details.Payload.Status.State, "expected replication status to be READY") {
+				t.Run("deactivate tenant", func(t *testing.T) {
+					tenants := []*models.Tenant{{Name: tenantName, ActivityStatus: "COLD"}}
+					helper.UpdateTenants(t, paragraphClass.Class, tenants)
+				})
```

Comment:
why do we keep deactivating the tenant until the replication status reaches READY? i was expecting this test to only have the deactivation call on line 267

Reply:
intentionally to trigger the deactivation on the tenant